src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -459,17 +461,29 @@ private static void LookupMembersInNamespace(LookupResult result, NamespaceSymbo
         /// </summary>
         private void LookupExtensionMethodsInSingleBinder(ExtensionMethodScope scope, LookupResult result, string name, int arity, LookupOptions options, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
         {
-            var methods = ArrayBuilder<MethodSymbol>.GetInstance();
-            var binder = scope.Binder;
-            binder.GetCandidateExtensionMethods(methods, name, arity, options, this);
-
+            var methods = scope.Binder.GetAllExtensionMethods();
```

Comment:
This might do more work, depending on the given `options`, I think.

@cston @jcouv Any ideas what I should do here?

Reply:
To give a bit of background, when binding a deeply nested expression that involves binding lots of extension methods, it seems that getting extension methods is done repeatedly over and over again for each call. So I thought this should be calculated once and cached in the binder. Am I moving in the right direction?

Reply:
I've adjusted the caching to work only for `arity == 0 && options == LookupOptions.AllMethodsOnArityZero` which is the very common call in my case.

Reply:
> when binding a deeply nested expression that involves binding lots of extension methods, it seems that getting extension methods is done repeatedly over and over again for each call

Is the scenario deeply nested *lamda expressions with overloaded methods*? If so, perhaps a good fraction of the time is binding lambda bodies for nested error cases, where the containing method call (with the lambda argument) is not applicable. That was the motivation for https://github.com/dotnet/roslyn/pull/59733.

Reply:
I have no idea if #59733 will help. The baseline code I'm starting with has no errors, but definitely as I start typing in the IDE there will be some errors (especially in the very inner invocations). @cston Should I run RoslynDeployment from #59733 locally and just check there if it fixes the issue?

Will #59733 be ready to merge anytime soon?

Reply:
> The baseline code I'm starting with has no errors

If the scenario involves nested lambda expressions and overloads with distinct delegate signatures, then even when compiling successfully, some inapplicable overloads may result in binding lambda bodies with errors.

Reply:
Then maybe #59733 will have a good impact. Should I expect it to get merged soon?